pkg/util/log/middleware.go

```diff
@@ -2,22 +2,41 @@ package log
 
 import (
 	"bufio"
+	"context"
 	"fmt"
 	"net"
 	"net/http"
 	"strings"
 	"time"
 
 	kitlog "github.com/go-kit/kit/log"
+	uuid "github.com/satori/go.uuid"
 )
 
+const (
+	KubernikusRequestID = "Kubernikus-Request-Id"
+)
+
+func RequestIDHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, request *http.Request) {
+		if id := request.Context().Value(KubernikusRequestID); id == nil {
+			request = request.WithContext(context.WithValue(request.Context(), KubernikusRequestID, uuid.NewV4()))
+		}
+		next.ServeHTTP(rw, request)
+	})
+}
+
 func LoggingHandler(logger kitlog.Logger, next http.Handler) http.Handler {
-	logger = kitlog.With(logger, "api", "ingress")
+	ingress_logger := kitlog.With(logger, "api", "ingress")
 	return http.HandlerFunc(func(rw http.ResponseWriter, request *http.Request) {
 		wrapper := makeWrapper(rw)
 
+		id := fmt.Sprintf("%s", request.Context().Value(KubernikusRequestID))
+		request = request.WithContext(context.WithValue(request.Context(), "logger", kitlog.With(logger, "id", id)))
+
```

Comment:
I know it annoying to handle this but we should handle the case were the there is no kubernikus request id in the content (because the requestid middleware is disabled).

Reply:
I'm omitting the `id` key now if the context does not contain the requestId.